web/federate.go

```diff
@@ -0,0 +1,98 @@
+package web
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"bitbucket.org/ww/goautoneg"
+	"github.com/golang/protobuf/proto"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/text"
+
+	"github.com/prometheus/prometheus/promql"
+
+	clientmodel "github.com/prometheus/client_golang/model"
+	dto "github.com/prometheus/client_model/go"
+)
+
+type Federation struct {
+	QueryEngine *promql.Engine
+}
+
+func (fed *Federation) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	req.ParseForm()
+	qs := fmt.Sprintf("{%s}", strings.Join(req.Form["match[]"], ","))
```

Comment:
For context: routing this through the engine is not ideal. We need some changes to the storage that will save us a lot of unnecessary processing. It will certainly not end in a release in this form.

ORing needs another query format. Combining AND and OR seems a bit much, so maybe `type=AND|OR` as an additional parameter?